scripts/o3de/o3de/repo_properties.py

```diff
@@ -92,16 +94,52 @@ def create_remote_object_archive(src_data_path: pathlib.Path,
         FileNotFoundError: If the json_data_path does not exist.
         ValueError: If the json_data_path is not a dict.
     """
-    if download_prefix is None:
-        logger.error('The --download-prefix argument must be provided. A url prefix for a file attached to a Github release might look like this:'
-                     '-dp https://github.com/o3de/o3de-extras/releases/download/2305.0')
-        return {}
     zip_path = releases_path / archive_filename
     # check if a object.zip folder already exist in the path - ask user if they want to overwrite the current zip
     if not force and zip_path.exists():
         logger.error(f'{zip_path} already exists.  Use --force command to overwrite the existing zip or '
                      'provide a new location to save your archive.')
         return {}
+    if not upload_git_release_tag and not download_prefix:
+        logger.error('You need to provide a --upload-github-release-tag if you want to upload to github.\n'
+                     'If you want to use other version control systems you must provide --download-prefix argument\n'
+                     'A url prefix for a file attached to a Github release might look like this:'
+                     '-dp https://github.com/o3de/o3de-extras/releases/download/2305.0')
+        return {}
+    if upload_git_release_tag and download_prefix:
+        logger.error('Upload to GitHub will automatically generate a download prefix for you.\n'
```

Comment:
In the future, we need to consider if this is the case for other Git providers besides GitHub
```suggestion
        logger.error('When a Git release tag is provided, no download prefix should be provided because  the download prefix is automatically generated.\n'
```